src/code/z_camera.c

```diff
@@ -28,33 +23,52 @@ s32 Camera_CheckWater(Camera* camera);
 /*===============================================================*/
 
 /**
- * Interpolates along a curve between 0 and 1 with a period of
- * -a <= p <= a at time `b`
+ * Interpolates along the curve shown below
+ * returns value y ranging from 0.0f to 1.0f for -xMax <= x <= xMax
+ * returns 1.0f otherwise
+ *
+ * y = 1.0f    ________                   _________
+ *                     __               __
+ *                       _             _
+ * y axis                 _           _
+ *                         ___     ___
+ *                            _____
+ * y = 0.0f           |         |         |
+ *                 -xMax        0       xMax
+ *
+ *                           x axis
  */
-f32 Camera_InterpolateCurve(f32 a, f32 b) {
-    f32 ret;
-    f32 absB;
-    f32 t = 0.4f;
-    f32 t2;
-    f32 t3;
-    f32 t4;
-
-    absB = fabsf(b);
-    if (a < absB) {
-        ret = 1.0f;
+f32 Camera_InterpolateCurve(f32 xMax, f32 x) {
+    f32 y;
+    f32 absX = fabsf(x);
+    f32 percent40 = 0.4f;
+    f32 percent60;
+    f32 xQuadratic;
+    f32 xMaxQuadratic; // Normalizing constant
+
+    // fixed value outside xMax range
+    if (absX > xMax) {
+        y = 1.0f;
+
+        // inside xMax range
     } else {
-        t2 = 1.0f - t;
-        if ((a * t2) > absB) {
-            t3 = SQ(b) * (1.0f - t);
-            t4 = SQ(a * t2);
-            ret = t3 / t4;
+        percent60 = 1.0f - percent40;
+
+        // quadratic curve in the inner 60% of xMax range: +concavity (upward curve)
+        if (absX < (xMax * percent60)) {
+            xQuadratic = SQ(x) * (1.0f - percent40);
+            xMaxQuadratic = SQ(xMax * percent60);
+
+            y = xQuadratic / xMaxQuadratic;
+            // quadratic curve in the outer 40% of xMax range: -concavity (flattening curve)
         } else {
-            t3 = SQ(a - absB) * t;
-            t4 = SQ(0.4f * a);
-            ret = 1.0f - (t3 / t4);
+            xQuadratic = SQ(xMax - absX) * percent40;
```

Comment:
I pulled the term concave from calculus (as a representation of second derivatives if you're familiar with that), basically its a property of a curve if it's slowly bending upward or slowly bending downward. It's different than just downward (negative first derivative), basically it's a piecewise function, with +x^2 in the middle 60% and -x^2 in the outer 40%. I'll try to better explain that property without concavity (since yes, it could be mixed up with concavity/convexity)